### PR TITLE
Do not follow symlinks or junctions when recursively deleting files

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/213)

### CODE REVIEW

File: utils/src/main/kotlin/Utils.kt [hunk 1]
```diff
@@ -140,10 +146,35 @@ fun String.fileSystemDecode(): String =
         java.net.URLDecoder.decode(this, "UTF-8")
 
 /**
- * Delete files recursively without failing if e.g. individual files in symlinked directories could not be deleted due
- * to permission issues if in the end the named directory does not exist anymore.
+ * Delete files recursively without following symbolic links (Unix) or junctions (Windows).
```
Comment (mnonnenmacher) [inline utils/src/main/kotlin/Utils.kt]: I don't understand how the code below skips symbolic links, could you explain please?

Comment (sschuberth) [reply utils/src/main/kotlin/Utils.kt]: It's implicitly documented in the Java SDK. I'll add a code comment.

**Comment by sschuberth** [comment]
Ok, I have no idea why AppVeyor gives

    com.here.ort.analyzer.integration.GradleIntegrationTest > executionError FAILED
        java.nio.file.AccessDeniedException: C:\Users\appveyor\AppData\Local\Temp\tmp5520974503866671023.tmp\Gradle\4.4.0\.git\objects\pack\pack-f4bb5bb991fc04bb4eb0b4664a633cc4eba1b8d4.idx

I cannot reproduce these locally.

File: utils/src/main/kotlin/Utils.kt (hunk 1 above)
Comment (mnonnenmacher) [reply utils/src/main/kotlin/Utils.kt]: Thanks.

**Comment by mnonnenmacher** [comment]
We should check if this happens for all builds now or if it is caused by this PR. I'll upload a fixed version of #211 soon, let's see if it fails with the same error.

**Comment by sschuberth** [comment]
Luckily the stack trace is [directly available from AppVeyor](https://ci.appveyor.com/project/heremaps/oss-review-toolkit/build/1.0.1308/tests):

    java.nio.file.AccessDeniedException: C:\Users\appveyor\AppData\Local\Temp\tmp8994948545899642444.tmp\OSS+Review+Toolkit\.git\objects\pack\pack-3172494e6c981340a4c0c3b0cd5aa5b3dde3ed66.idx
        at sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:83)
        at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:97)
        at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:102)
        at sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:269)
        at sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:103)
        at java.nio.file.Files.delete(Files.java:1126)
        at com.here.ort.utils.UtilsKt$safeDeleteRecursively$1.visitFile(Utils.kt:172)
        at com.here.ort.utils.UtilsKt$safeDeleteRecursively$1.visitFile(Utils.kt:158)
        at java.nio.file.Files.walkFileTree(Files.java:2670)
        at java.nio.file.Files.walkFileTree(Files.java:2742)
        at com.here.ort.utils.UtilsKt.safeDeleteRecursively(Utils.kt:158)
        at com.here.ort.analyzer.integration.AbstractIntegrationSpec.interceptSpec(AbstractIntegrationSpec.kt:74)

So it indeed happens in my new code.

**Comment by sschuberth** [comment]
But I guess this in fact already happened before, but before we did not check the return value of `outputDir.deleteRecursively()`. The new implementation throws an exception if a file could not be deleted.

Sill wondering why this only occurs for tests derived from `AbstractIntegrationSpec`, not for any of the other tests that clone a Git repo.

**Comment by sschuberth** [comment]
Ha, I can reproduce it locally if I run the test on the command line via

    ./gradlew analyzer:funTest --tests com.here.ort.analyzer.integration.GradleIntegrationTest

Running it from the IDE works.

**Comment by sschuberth** [comment]
Turns out the error *did* occur in the IDE, but the tests were still all marked as green. Still the exception could be seen in the console output of the *top-level* test entry. Anyway, I have a fix now 😁

**Pull request merged**